### PR TITLE
WIP: Function to modify gromacs PDB to have unique chain IDs

### LINKED ISSUE #11: Changing gromacs generated PDB chains to work for plumed

### PR DESCRIPTION
**Pull request by hgandhi2411**
Resolves #11 
This PR adds peptidesim functions to use pdb files for the following purposes:
* get the atom count in each chain (excludes solvent molecules).
* remove solvent from pdb file.
* rename chain ids to have unique letters in the pdb, for plumed related simulations. 

I do not have tests for these functions. So, to review, please use any pdb files that you may already have as part of your simulations or ask me to send one to you for testing purposes.

### CODE REVIEW

**Comment by whitead** [comment]
Can you remove the pickling as long as you're editing this file to reflect the new pickling syntax?

File: package/peptidesim/utilities.py [hunk 1]
```diff
@@ -108,3 +108,153 @@ def parser(plumed_file):
...
+        pdb_for_plumed(input_file='template.pdb',
+                        peptide_copies=[6,2],
+                        atoms_in_chain=[35,43],
+                        first_atom_index=5,
+                        output_file='new.pdb')
+    '''
+    from string import ascii_uppercase
+
+    # read the pdb file
+    with open(input_pdbfile, 'r') as f:
```
Comment (dilnoza92) [inline package/peptidesim/utilities.py:145]: the keyword argument is input_file not input_pdbfile.

File: inputs/abeta_sims/part_1.py [hunk 2]
```diff
@@ -1,197 +1,54 @@
...
-                    f.write('ENDMDL second\n')  # puts finish touches
-                    f.close()  # done
-    return output_pdbfile
-
-
-file3 = 'template.pdb'
-new_pdb_file = pdbfile_generator_w_chain_id(
-    peptide_copies, atoms_in_chain, 5, 'new_pdb.pdb', file0)
+plumed_pdb = 'template.pdb'
+new_pdb_file = utilities.pdb_for_plumed(input_pdbfile=file0,
```
Comment (dilnoza92) [inline inputs/abeta_sims/part_1.py:40]: the first argument should be input_file

File: inputs/abeta_sims/part_1.py [hunk 3]
```diff
@@ -1,197 +1,54 @@
...
+                                        peptide_copies=peptide_copies,
+                                        atoms_in_chain=atoms_in_chain,
+                                        first_atom_index=5,
+                                        output_pdbfile=plumed_pdb)
```
Comment (dilnoza92) [inline inputs/abeta_sims/part_1.py:44]: again this should be output_file

**Comment by dilnoza92** [comment]
@mehradans92: when i tried running test_utilities.py, i got the following error   "/package/tests/test_utilities.py:13: DeprecationWarning: invalid escape sequence \.
    txt_plumed = ('PRINT ARG=(cs\.cb_.*),(cs\.ca_.*),(cs\.hn_.*),(cs\.ha_.*),(cs\.nh_.*),(cs\.co_.*) FILE=WT_PTE_CS_shifts STRIDE=250\n' +"



The plumed 2.5 has this syntax "PRINT ARG=(cs\.cb-.*),(cs\.ca-.*),(cs\.hn-.*),(cs\.ha-.*),(cs\.nh-.*),(cs\.co-."

**Comment by mehradans92** [comment]
@dilnoza92 . How did you run the test?

**Comment by dilnoza92** [comment]
@mehradans92 I typed `python -m pytest test_utilities.py -v`  in the tests folder in peptidesim

**Comment by mehradans92** [comment]
@dilnoza92 Could you try `python test_utils.py`?

**Comment by hgandhi2411** [comment]
@dilnoza92 thanks for pointing out the error. I have fixed all errors and tested on an example system. Can you check to see that it works for now? Thanks.

**Pull request merged**